### PR TITLE
Upgrade Ember from 3.10 to 3.19 include some incidental changes

### PR DESCRIPTION
**Pull request by allthesignals**
Addresses #1058 by upgrading Ember to 3.19.

Current WIP, description coming...

### CODE REVIEW

File: package.json
Comment (allthesignals) [inline package.json:66]: generated by https://github.com/ember-cli/ember-cli-update

File: package.json
Comment (allthesignals) [inline package.json:93]: generated by https://github.com/ember-cli/ember-cli-update

File: package.json
Comment (allthesignals) [inline package.json:110]: I removed this as a temporary convenience but happy to re-add it. As far as I could tell it was not working. If it's valuable to the team still then we can add it but it seems redundant to the current CI steps.

File: testem.js
Comment (allthesignals) [inline testem.js:18]: generated by https://github.com/ember-cli/ember-cli-update

note that the updater will try to change many things about this file. however, this is configured to work for testing mapbox-gl in a headless environment. so we should be careful with this.

File: .gitignore
Comment (allthesignals) [inline .gitignore:28]: Meant for debugging dependency issues using https://github.com/wclr/yalc

File: package.json
Comment (allthesignals) [inline package.json:21]: The project won't build on my machine without this dependency. I'm not sure if some version drift dropped explicit inclusion of this (and everyone just has a stale node_modules on their machine), but I had to include this dependency.

File: .nvmrc
Comment (TylerMatteo) [reply .nvmrc:1]: I was actually going to ask you prefer `.nvm` over `.node-version` anyway so this is great! I know that Netlify also honors both of them when its trying to figure out which node version to build with.

File: app/adapters/lot.js
Comment (TylerMatteo) [inline app/adapters/lot.js:52]: Can you explain why you're removing this? If it really should be removed, let's just delete the line instead of commenting

File: app/components/mapbox/basic-map.js
Comment (TylerMatteo) [inline app/components/mapbox/basic-map.js:17]: Can you explain what this was doing and why it's ok to delete it? In general, prefer deleting over commenting out.

File: package.json
Comment (TylerMatteo) [reply package.json:113]: If it's causing huge headaches, I'm cool with deleting this for the foreseeable future but ideally this would be something we remove only temporarily. Hopefully just upgrading to a newer version of `lint-staged` helps.

File: tests/acceptance/bbox-test.js
Comment (TylerMatteo) [inline tests/acceptance/bbox-test.js:31]: Remove this?

File: tests/acceptance/bookmarks-test.js
Comment (TylerMatteo) [inline tests/acceptance/bookmarks-test.js:60]: Can you explain what Percy is and why you're removing stuff from this test? I don't mind changing tests to get these upgrades done (especially if the tests you're changing weren't particularly valuable in the first place), I just want to make sure we have some context documenting in GH

File: package.json
Comment (TylerMatteo) [reply package.json:110]: It is redundant with CI but it's intentional redundancy 😄 that being said, fine with removing it until we get things upgraded and then revisit it. I think you're correct that it's not quite set up correctly anyway.

**Review by TylerMatteo — CHANGES_REQUESTED** [review]

File: app/adapters/lot.js
Comment (TylerMatteo) [reply app/adapters/lot.js:52]: Don't feel like I _totally_ understand what's going on here but if you're confident that this line isn't affecting anything outside of when tests are running, I'm happy to defer to your Ember expertise here.

File: package.json
Comment (TylerMatteo) [inline package.json:122]: Can you change this to `^12`? I think it would be best to keep these to one major version at a time.

File: package.json
Comment (TylerMatteo) [inline package.json:76]: You should be able to change this to `0.3.0` now. I published a [new version](https://www.npmjs.com/package/ember-mapbox-composer/v/0.3.0) on 11/3 that has the changes from your most recent PR in that repo. Let's pin this dependency to exactly that version.

File: app/components/labs-map.js
Comment (TylerMatteo) [inline app/components/labs-map.js:7]: Can you remind me why you need this new file? Is this "extending" the labs-map component that's coming from the addon?

**Review by TylerMatteo — CHANGES_REQUESTED** [review]

**Comment by TylerMatteo** [comment]
@allthesignals How is going with re-enabling the git hooks? I saw that you have a new commit so I pulled your branch and tried to commit a change (running Node 14.18.1) and got this error
```
❯ git commit -a -m "Testing hooks"
yarn run v1.22.19
$ /Users/tylermatteo/work/ose_repos/labs-zola/node_modules/.bin/lint-staged
✔ Preparing lint-staged...
❯ Running tasks for staged files...
  ❯ package.json — 1 file
    ❯ *.js — 1 file
      ✖ eslint --cache --fix [FAILED]
↓ Skipped because of errors from tasks. [SKIPPED]
✔ Reverting to original state because of errors...
✔ Cleaning up temporary files...

✖ eslint --cache --fix:

Oops! Something went wrong! :(

ESLint: 7.32.0

Error: Failed to load parser 'babel-eslint' declared in '.eslintrc.js': Cannot find module 'babel-eslint'
Require stack:
- /Users/tylermatteo/work/ose_repos/labs-zola/.eslintrc.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at Function.resolve (internal/modules/cjs/helpers.js:99:19)
    at Object.resolve (/Users/tylermatteo/work/ose_repos/labs-zola/node_modules/@eslint/eslintrc/lib/shared/relative-module-resolver.js:28:50)
    at ConfigArrayFactory._loadParser (/Users/tylermatteo/work/ose_repos/labs-zola/node_modules/@eslint/eslintrc/lib/config-array-factory.js:929:45)
    at ConfigArrayFactory._normalizeObjectConfigDataBody (/Users/tylermatteo/work/ose_repos/labs-zola/node_modules/@eslint/eslintrc/lib/config-array-factory.js:724:43)
    at _normalizeObjectConfigDataBody.next (<anonymous>)
    at ConfigArrayFactory._normalizeObjectConfigData (/Users/tylermatteo/work/ose_repos/labs-zola/node_modules/@eslint/eslintrc/lib/config-array-factory.js:665:20)
    at _normalizeObjectConfigData.next (<anonymous>)
    at ConfigArrayFactory.loadInDirectory (/Users/tylermatteo/work/ose_repos/labs-zola/node_modules/@eslint/eslintrc/lib/config-array-factory.js:511:28)
    at CascadingConfigArrayFactory._loadConfigInAncestors (/Users/tylermatteo/work/ose_repos/labs-zola/node_modules/@eslint/eslintrc/lib/cascading-config-array-factory.js:379:46)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for docu
```

**Review by TylerMatteo — APPROVED** [review]
@allthesignals  Tests and commit hooks working as expected for me locally! Shall we merge? I'll figure out what QA is going to look like on our end to get this to master next.

**Comment by TylerMatteo** [comment]
@allthesignals  Actually the tests are failing when I retried this morning. Lets discuss in our call tomorrow.

File: package.json
Comment (TylerMatteo) [inline package.json:15]: Hey @allthesignals I was just looking this PR over one more time before merging and noticed this. Can you explain why you needed this change? I see that the `lint-staged` section in package.json is running `eslint --cache --fix` so this probably isn't relevant to that but just running `yarn run lint` locally spit out a bunch of errors.

Does this change allow things through the linter that weren't allowed through before? I'm not necessarily opposed to that for this PR if we need to, just want to understand what it's doing.

Comment (TylerMatteo) [reply package.json:15]: Oh I just realized that this command is only linting the handlebars templates. The [package](https://www.npmjs.com/package/ember-template-lint) we use for that doesn't seem to have a `fix` option but I'm not as concerned about those files anyways. Thanks for the context.

**Pull request merged**
